### PR TITLE
Add relevant data to Site Health screen

### LINKED ISSUE #4139: Add relevant data to Site Health screen
## Feature Description

To be able to better support our users, we need some data points in Site Health related to stories.

Filter `debug_information` to add:

* libxml version (see https://github.com/ampproject/amp-wp/blob/be84e8a2fb8def6b1831656d2067b268e70c959c/src/Admin/SiteHealth.php#L524-L528)
* `WEBSTORIES_DEV_MODE` value 
* `WEBSTORIES_CDN_URL` value
* List of active experiments

Filter `site_status_test_php_modules` to include `mbstring` (see https://github.com/ampproject/amp-wp/blob/be84e8a2fb8def6b1831656d2067b268e70c959c/src/Admin/SiteHealth.php#L661-L664)

Filter `site_status_test_result` to mark the `https_status` test status as critical if its current status is recommended (see https://github.com/ampproject/amp-wp/blob/be84e8a2fb8def6b1831656d2067b268e70c959c/src/Admin/SiteHealth.php#L535-L555)

## Additional Context

Here's how the AMP plugin does it:

https://github.com/ampproject/amp-wp/blob/dce58640f7c36befda67b145d00ec4f40c2278eb/src/Admin/SiteHealth.php#L447-L450

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief

* Create a new `Site_Health` class in the `Admin` namespace

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Added site health class. 

- Added some more data that requested. Can be removed if required. 
   -  Added more data to required extensions. This closer maps to compatibility class. 
   - Added `WEBSTORIES_VERSION`, `WEBSTORIES_DB_VERSION`, `WEBSTORIES_AMP_VERSION`, `WEBSTORIES_CDN_URL`  and `WEBSTORIES_DEV_MODE `to debug information
   - Added theme supports to debug information. 

## Relevant Technical Choices

Little unit test light on this one, as most of the changes are adding to filters. 

## User-facing changes

Screenshot 2021-05-19 at 09 47 50

Screenshot 2021-05-19 at 09 46 43

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Login as admin
2. Tools > Site health. 
3. Check test pass. 
4. Click on info. 
5. See Web stories constants and web stories Experiments sections. 

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #4139

### CODE REVIEW

File: includes/Admin/Site_Health.php [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+			],
+		];
+
+		foreach ( $this->experiments->get_experiments() as $experiment ) {
+			$extra_data['web_stories_experiments']['fields'][ $experiment['name'] ] = [
+				'label'   => $experiment['label'],
+				'value'   => $this->get_formatted_output( $this->experiments->is_experiment_enabled( $experiment['name'] ) ),
+				'private' => false,
+			];
+		}
```
Comment (swissspidy) [inline includes/Admin/Site_Health.php]: Using `$this->experiments->get_enabled_experiments()` makes the list a bid tidier.

Not interested in disabled ones as they don't have any effect.

```suggestion
		foreach ( $this->experiments->get_enabled_experiments() as $experiment ) {
			$extra_data['web_stories_experiments']['fields'][ $experiment['name'] ] = [
				'label'   => $experiment['label'],
				'value'   => $this->get_formatted_output( true ),
				'private' => false,
			];
		}
```

File: includes/Admin/Site_Health.php [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+						'private' => false,
+					],
+					'web_stories_libxml_version' => [
+						'label'   => 'libxml Version',
+						'value'   => LIBXML_DOTTED_VERSION,
+						'private' => false,
+					],
+				],
+			],
+			'web_stories_experiments' => [
```
Comment (swissspidy) [inline includes/Admin/Site_Health.php]: Can we make this part of the above section instead?

I don't want to clutter Site Health with two large sections for one plugin. Combining them would keep things a bit tidier.

Then the field label would be "Enabled experiments" and the enabled experiments would be listed on the right in a `<ul>`.

Mock:

Screenshot 2021-05-19 at 01 26 28

File: includes/Admin/Site_Health.php [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+	 * @since 1.8.0
+	 *
+	 * @param array $debugging_information The debugging information from Core.
+	 *
+	 * @return array The debugging information, with added information for Web stories.
+	 */
+	public function add_debug_information( array $debugging_information ) {
+		$extra_data = [
+			'web_stories'             => [
+				'label'       => esc_html__( 'Web Stories Constants', 'web-stories' ),
```
Comment (swissspidy) [inline includes/Admin/Site_Health.php]:
```suggestion
				'label'       => esc_html__( 'Web Stories', 'web-stories' ),
```

**Review by swissspidy — COMMENTED** [review]
Only one question basically about grouping the data in 1 section, otherwise looks good 👍

File: includes/Admin/Site_Health.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/Admin/Site_Health.php]: We just want enabled experiments, is that correct?

Comment (swissspidy) [reply includes/Admin/Site_Health.php]: Yes 👍

File: includes/Admin/Site_Health.php (hunk 3 above)
Comment (spacedmonkey) [reply includes/Admin/Site_Health.php]: Fixed in https://github.com/google/web-stories-wp/pull/7570/commits/0b0e2cad6be248e8df3e18c5faaebc723f4ae53d

File: includes/Admin/Site_Health.php (hunk 2 above)
Comment (spacedmonkey) [reply includes/Admin/Site_Health.php]: It turns out that core support an [array](https://github.com/WordPress/wordpress-develop/blob/a32a192ede496cf0072ae957d2e780212a04b8f5/src/wp-admin/site-health-info.php#L108-L119) value. 

Without a little tweak, I got this working.

File: includes/Admin/Site_Health.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/Admin/Site_Health.php]: Tweaked in https://github.com/google/web-stories-wp/pull/7570/commits/0b0e2cad6be248e8df3e18c5faaebc723f4ae53d

**Comment by spacedmonkey** [comment]
Updated PR based on feedback. Updated screenshot.

File: includes/Admin/Site_Health.php [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+	 * @param array $debugging_information The debugging information from Core.
+	 *
+	 * @return array The debugging information, with added information for Web stories.
+	 */
+	public function add_debug_information( array $debugging_information ) {
+		$enabled_experiments = [];
+		foreach ( $this->experiments->get_experiments() as $experiment ) {
+			$enabled = $this->experiments->is_experiment_enabled( $experiment['name'] );
+			if ( $enabled ) {
+				$enabled_experiments[ $experiment['label'] ] = $this->get_formatted_output( $enabled );
```
Comment (swissspidy) [inline includes/Admin/Site_Health.php:111]:
```suggestion
				$enabled_experiments[] = $experiment['label'];
```

Does this work as well?

A bit repetitive to show "Enabled" for every experiment when we're only showing enabled experiments anyway.

Comment (swissspidy) [reply includes/Admin/Site_Health.php:111]: Ah of course it doesn't

https://github.com/WordPress/wordpress-develop/blob/a32a192ede496cf0072ae957d2e780212a04b8f5/src/wp-admin/site-health-info.php#L108-L119

Well then I suppose we'll leave it like that.

**Review by swissspidy — APPROVED** [review]

File: includes/Admin/Site_Health.php (hunk 4 above)
Comment (spacedmonkey) [reply includes/Admin/Site_Health.php:111]: Sorry, I should have explained this. Without a key it ends up looking like this. 

Screenshot 2021-05-19 at 12 13 21

Which kind doesn't look great.

**Comment by spacedmonkey** [comment]
I created a core ticket inspired by this PR - https://core.trac.wordpress.org/ticket/53230

File: includes/Admin/Site_Health.php [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+			}
+		}
+		if ( ! $enabled_experiments ) {
+			$enabled_experiments = __( 'No experiments enabled', 'web-stories' );
+		}
+
+		$extra_data = [
+			'web_stories' => [
+				'label'       => esc_html__( 'Web Stories', 'web-stories' ),
+				'description' => esc_html__( 'Debugging information for the Web Stories for WordPress.', 'web-stories' ),
```
Comment (swissspidy) [inline includes/Admin/Site_Health.php]:
```suggestion
				'description' => esc_html__( 'Debugging information for the Web Stories plugin.', 'web-stories' ),
```

Just noticed this by accident

Comment (swissspidy) [reply includes/Admin/Site_Health.php]: @spacedmonkey Can you commit this before merging? TIA

**Pull request merged**